src/hooks/useAsyncActionHandler/useAsyncActionHandler.ts

```diff
@@ -0,0 +1,59 @@
+import React from 'react';
+
+export interface UseAsyncActionHandlerProps<ActionResult = void> {
+    handler: (...args: any[]) => Promise<ActionResult>;
+}
+
+export interface UseAsyncActionHandlerResult<Action = () => Promise<void>> {
+    isLoading: boolean;
+    handler: Action;
+}
+
+export function useAsyncActionHandler<ActionResult>({
+    handler,
+}: UseAsyncActionHandlerProps<ActionResult>): UseAsyncActionHandlerResult<typeof handler> {
+    const mounted = React.useRef(true);
+    const [isLoading, setLoading] = React.useState(false);
+
+    React.useEffect(() => {
+        mounted.current = true;
+
+        return () => {
+            mounted.current = false;
+        };
+    }, []);
+
+    const handleAction: typeof handler = React.useCallback(
+        (...args) => {
+            setLoading(true);
+
+            const handlerResult = handler(...args);
+
+            if (handlerResult instanceof Promise) {
+                return handlerResult.finally(() => {
+                    if (mounted.current) {
```

Comment:
Why do we need this check?

Reply:
Not calling setState on unmount component?

Reply:
In React 18  the warning about using setState on an unmounted component has been removed as misleading. And this wrong solution is described here https://github.com/reactwg/react-18/discussions/82

Reply:
Didn't know that. Ok then let's remove this code